src/csync/csync.h

```diff
@@ -149,134 +157,51 @@ enum csync_ftw_type_e {
 // currently specified at https://github.com/owncloud/core/issues/8322 are 9 to 10
 #define REMOTE_PERM_BUF_SIZE 15
 
-typedef struct csync_vio_file_stat_s csync_vio_file_stat_t;
-
-enum csync_vio_file_flags_e {
-  CSYNC_VIO_FILE_FLAGS_NONE = 0,
-  CSYNC_VIO_FILE_FLAGS_SYMLINK = 1 << 0,
-  CSYNC_VIO_FILE_FLAGS_HIDDEN = 1 << 1
-};
-
-enum csync_vio_file_type_e {
-  CSYNC_VIO_FILE_TYPE_UNKNOWN,
-  CSYNC_VIO_FILE_TYPE_REGULAR,
-  CSYNC_VIO_FILE_TYPE_DIRECTORY,
-  CSYNC_VIO_FILE_TYPE_FIFO,
-  CSYNC_VIO_FILE_TYPE_SOCKET,
-  CSYNC_VIO_FILE_TYPE_CHARACTER_DEVICE,
-  CSYNC_VIO_FILE_TYPE_BLOCK_DEVICE,
-  CSYNC_VIO_FILE_TYPE_SYMBOLIC_LINK
-};
-
-enum csync_vio_file_stat_fields_e {
-  CSYNC_VIO_FILE_STAT_FIELDS_NONE = 0,
-  CSYNC_VIO_FILE_STAT_FIELDS_TYPE = 1 << 0,
-  CSYNC_VIO_FILE_STAT_FIELDS_MODE = 1 << 1, // local POSIX mode
-  CSYNC_VIO_FILE_STAT_FIELDS_FLAGS = 1 << 2,
-//  CSYNC_VIO_FILE_STAT_FIELDS_DEVICE = 1 << 3,
-  CSYNC_VIO_FILE_STAT_FIELDS_INODE = 1 << 4,
-//  CSYNC_VIO_FILE_STAT_FIELDS_LINK_COUNT = 1 << 5,
-  CSYNC_VIO_FILE_STAT_FIELDS_SIZE = 1 << 6,
-//  CSYNC_VIO_FILE_STAT_FIELDS_BLOCK_COUNT = 1 << 7, /* will be removed */
-//  CSYNC_VIO_FILE_STAT_FIELDS_BLOCK_SIZE = 1 << 8,  /* will be removed */
-  CSYNC_VIO_FILE_STAT_FIELDS_ATIME = 1 << 9,
-  CSYNC_VIO_FILE_STAT_FIELDS_MTIME = 1 << 10,
-  CSYNC_VIO_FILE_STAT_FIELDS_CTIME = 1 << 11,
-//  CSYNC_VIO_FILE_STAT_FIELDS_SYMLINK_NAME = 1 << 12,
-//  CSYNC_VIO_FILE_STAT_FIELDS_CHECKSUM = 1 << 13,
-//  CSYNC_VIO_FILE_STAT_FIELDS_ACL = 1 << 14,
-//  CSYNC_VIO_FILE_STAT_FIELDS_UID = 1 << 15,
-//  CSYNC_VIO_FILE_STAT_FIELDS_GID = 1 << 16,
-  CSYNC_VIO_FILE_STAT_FIELDS_ETAG = 1 << 17,
-  CSYNC_VIO_FILE_STAT_FIELDS_FILE_ID = 1 << 18,
-  CSYNC_VIO_FILE_STAT_FIELDS_DIRECTDOWNLOADURL = 1 << 19,
-  CSYNC_VIO_FILE_STAT_FIELDS_DIRECTDOWNLOADCOOKIES = 1 << 20,
-  CSYNC_VIO_FILE_STAT_FIELDS_PERM = 1 << 21 // remote oC perm
-
-};
-
-
-struct csync_vio_file_stat_s {
-  char *name;
-  char *etag; // FIXME: Should this be inlined like file_id and perm?
-  char file_id[FILE_ID_BUF_SIZE+1];
-  char *directDownloadUrl;
-  char *directDownloadCookies;
-  char remotePerm[REMOTE_PERM_BUF_SIZE+1];
-
-  time_t atime;
-  time_t mtime;
-  time_t ctime;
+typedef struct csync_file_stat_s csync_file_stat_t;
 
+struct OCSYNC_EXPORT csync_file_stat_s {
+  uint64_t phash;
+  time_t modtime;
   int64_t size;
-
-  mode_t mode;
-
   uint64_t inode;
-
-  int fields; // actually enum csync_vio_file_stat_fields_e fields;
-  enum csync_vio_file_type_e type;
-
-  int flags;
-
-  char *original_name; // only set if locale conversion fails
-
-  // For remote file stats: the highest quality checksum the server provided
-  // in the "SHA1:324315da2143" form.
-  char *checksumHeader;
+  enum csync_ftw_type_e type BITFIELD(4);
+  bool child_modified BITFIELD(1);
```

Comment:
Ony the enum needs the BITFIELD macro.   The bool can stay 1.
But that's not su important because it is only for old compiler anyway.